### ISSUE TITLE
IOError: cannot identify image file <StringIO.StringIO instance 

### ISSUE THREAD

**Issue by dakmatt** [tcassou/mapsplotlib#3]
Dears,

I've install mapsplotlib and try to run the following,

![screenshot from 2017-11-21 12-26-26](https://user-images.githubusercontent.com/5714914/33054903-4491dfb2-ceb7-11e7-829e-402a834da489.png)

However, got the following.

IOError: cannot identify image file <StringIO.StringIO instance 

I believe it was due to Pillow module didn't recognize certain image format.

Kindly advise.

**Comment by tcassou** [comment]
Hi @dakmatt,
Thanks for sharing, I tried to reproduce with the following data based on your screenshot:
```python
df = pd.DataFrame(
    [
        ["JUMLAH", 790136, np.nan, np.nan],
        ["ALOR GAJAH", 173712, 2.382211, 102.211561],
        ["Ayer Pa'Abas", 2644, 2.396907, 102.166641],
    ],
    columns=['label', 'size', 'latitude', 'longitude'],
)

mplt.plot_markers(df)
```
and it worked... except for the fact that the `NaN` values lead to a weird marker on the map, I'll add that improvement to the backlog:
![image](https://user-images.githubusercontent.com/8159430/33086427-da4e05d2-cee7-11e7-8d37-021ba22b4842.png)

Could you share more so that I can reproduce your error? 
I believe it is due to an unsuccessful query to Google APIs, which then return a wrong string (not an image basically) and Pillow fails to load it. Can't be 100% sure given the context though.

If I can reproduce it, I'm more than happy to add some verbose to the error, I acknowledge it's not very helpful at the moment!

Hope it helps,

Thomas

**Comment by tcassou** [comment]
Added this issue to the board https://github.com/tcassou/mapsplotlib/issues/4

**Comment by tcassou** [comment]
Any luck @dakmatt ?

**Comment by rudolfbyker** [comment]
I can confirm that this happens when Google sends a "403" response code. For me, it meant that I forgot to enable the "static maps" API.

**Comment by tcassou** [comment]
Ok thanks @rudolfbyker, I'll add some logic to handle the response codes.

**Comment by tcassou** [comment]
I released [version 1.0.6](https://github.com/tcassou/mapsplotlib/releases/tag/1.0.6) on pypi with HTTP error handling, it should prevent this issue in the future.
Thanks for reporting!
